dev/com.ibm.ws.transport.http/src/com/ibm/ws/genericbnf/internal/BNFHeadersImpl.java

```diff
@@ -1027,7 +1046,7 @@ public boolean containsHeader(HeaderKeys key) {
         if (null == key) {
             throw new IllegalArgumentException("Null input provided");
         }
-        return (null != findHeader(key));
+        return (headersAdded.get(key.getOrdinal()) == false ? false : (null != findHeader(key)));
```

Comment:
Why do the findHeader at all?

Reply:
That logic is backwards ... it should be returning true if headersAdded says the key is present, and calling findHeader(key) only if headersAdded doesn't know about the key.

findHeader(key) will look in the actual key storage, and will check for headers which go beyond the 1024 limit on headerKeys, so it will do a more complete search for the key

Hmm, think I need a private method to act as a wrapper when we check headersAdded for keys, to skip the check if the key ordinal exceeds headersAdded size (that will throw an exception)